x-pack/plugins/observability/public/components/shared/exploratory_view/exploratory_view.tsx

```diff
@@ -18,25 +18,25 @@ import { EmptyView } from './components/empty_view';
 import { TypedLensByValueInput } from '../../../../../lens/public';
 import { useAppIndexPatternContext } from './hooks/use_app_index_pattern';
 import { SeriesBuilder } from './series_builder/series_builder';
-import { AppDataType, SeriesUrl } from './types';
+import { SeriesUrl } from './types';
 
-const combineTimeRanges = (allSeries: Record<string, SeriesUrl>) => {
+export const combineTimeRanges = (
+  allSeries: Record<string, SeriesUrl>,
+  firstSeries?: SeriesUrl
+) => {
   let to: string = '';
   let from: string = '';
-  const allKeys = Object.keys(allSeries ?? {});
-  allKeys.forEach((seriesId) => {
-    const series = allSeries[seriesId];
+  if (firstSeries?.reportType === 'kpi-over-time') {
```

Comment:
so the way comparison works is that in case of timeseries/kpi over time, we always keep the same time window, but user can only do offset by selecting a different time window. 

But the length of the x-axis is always kept same. But in other serise types like for duration, it doesn't matter what windows, what time period they choose, we just add those as filters and extends the over all time range to incorporate all intervals.

Happy to jump on a call to explain further.